runtime/interpreter/interpreter.go

```diff
@@ -246,7 +246,7 @@ type Storage interface {
 	CheckHealth() error
 }
 
-type ReferencedResourceKindedValues map[atree.StorageID]map[ReferenceTrackedResourceKindedValue]struct{}
+type ReferencedResourceKindedValues map[atree.StorageID]map[*EphemeralReferenceValue]struct{}
```

Comment:
Do we not need to track invalidation of `StorageReferenceValue`s?

Reply:
No, storage references are always pointing to the same storage location. Storage references don't keep the value inside the reference (like ephemeral references do), rather always looked up from the storage, every time the value is used. So if the value is changed in the storage, it will automatically reflect when the storage look-up happens.

Reply:
Do you mean bound functions for storage references? In that case - yes.
(added a test https://github.com/onflow/cadence/pull/2958/commits/bfdd84a72a72cea758cf310be50069e3de562e60 to verify)